### ISSUE TITLE
Allow platform-dependence in targets?

### ISSUE THREAD

**Issue by GPMueller** [Trick-17/clang-build#68]
This would allow a lot of abstraction layers, such as GLFW, where the platform-dependence cannot be avoided, to be built quite naturally with `clang-build`. The goal should again be that the easiest solution is to have no platform-dependence, but to allow it if actually necessary.
Aside from abstraction layers, one sometimes has to apply platform-specific linkage, e.g. to the `pthreads` library on unix systems.

The following would allow it to be composable and non-intrusive:
```toml
[mytarget]
target_type = "shared library"

sources = ["src/common.cpp"]
[mytarget.flags]
compile = ["-DSOMETHING"]

[mytarget.osx]
sources = ["src/osx/platform_specific.cpp"]
[mytarget.osx.flags]
link = ['-lpthread']

[mytarget.linux]
sources = ["src/linux/platform_specific.cpp"]
[mytarget.linux.flags]
link = ['-lpthread']

[mytarget.windows]
sources = ["src/windows/platform_specific.cpp"]
```
The corresponding `osx`, `linux` and `windows` subdicts would be merged into the main one if the corresponding platform is detected. This of course only makes sense for lists, i.e. `include_directories`, `sources` and `flags` (*what about `dependencies`?*).

**Comment by GPMueller** [comment]
Implemented correctly with 2d54e0a5ff0dbd391b4112634cd2cecc4e1ab6df on `feature-improved-config`.

**Comment by GPMueller** [comment]
Added `public-include-directories` with ff0cfe4cb293c8b8e4a81781ea34ba84e228827b. Maybe we can think up something better or more intuitive, but it's definitely necessary to distinguish between include directories, which should or should not be inherited by dependent targets.
If it exists, the `include` directory of a target folder is automatically public.
